houston/root_cause/delta_debugging.py

```diff
@@ -15,14 +15,17 @@ class DeltaDebugging(RootCauseFinder):
     it down to main problem.
     """
     def __init__(self,
-                 system: System,
+                 system: Type[System],
                  initial_state: State,
                  environment: Environment,
                  config: Configuration,
-                 initial_failing_missions: List[Mission]
+                 initial_failing_missions: List[Mission],
+                 snapshot: str
                  ) -> None:
         self.__domain = MissionDomain.from_initial_mission(
             initial_failing_missions[0], discrete_params=True)
+        self.__bz = BugZoo()
```

Comment:
So, this is a little frowned upon because starting the BugZoo server can take quite a bit of time (roughly a minute) and introduces redundant memory consumption. This will become an issue if you're creating multiple instances of this class in a single execution. I would prefer that a BugZoo client/server be passed in the constructor, but it's not essential for now.